### PR TITLE
Makefile: automate creating multi-arch Docker manifests

### PR DESCRIPTION
**Pull request by invidian**
So single image reference can be used for all platforms.

Signed-off-by: Mateusz Gozdek <mateusz@kinvolk.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,6 +38,30 @@ build-docker-all-push: build-docker-all
...
+		--amend $(IMAGE_REPOSITORY)/bootkube:$(VERSION)-amd64 \
+		--amend $(IMAGE_REPOSITORY)/bootkube:$(VERSION)-arm64
+
+	docker manifest annotate $(IMAGE_REPOSITORY)/bootkube:$(VERSION) \
+		$(IMAGE_REPOSITORY)/bootkube:$(VERSION)-amd64 --arch amd64
+
+	docker manifest annotate $(IMAGE_REPOSITORY)/bootkube:$(VERSION) \
+		$(IMAGE_REPOSITORY)/bootkube:$(VERSION)-arm64 --arch arm64
+
+	docker manifest push $(IMAGE_REPOSITORY)/bootkube:$(VERSION)
```
Comment (knrt10) [inline Makefile:51]: can we please add `--purge` tag also, to delete the manifest locally.

File: Makefile [hunk 2]
```diff
@@ -38,6 +38,30 @@ build-docker-all-push: build-docker-all
...
+		--amend $(IMAGE_REPOSITORY)/checkpoint:$(VERSION)-amd64 \
+		--amend $(IMAGE_REPOSITORY)/checkpoint:$(VERSION)-arm64
+
+	docker manifest annotate $(IMAGE_REPOSITORY)/checkpoint:$(VERSION) \
+		$(IMAGE_REPOSITORY)/checkpoint:$(VERSION)-amd64 --arch amd64
+
+	docker manifest annotate $(IMAGE_REPOSITORY)/checkpoint:$(VERSION) \
+		$(IMAGE_REPOSITORY)/checkpoint:$(VERSION)-arm64 --arch arm64
+
+	docker manifest push $(IMAGE_REPOSITORY)/checkpoint:$(VERSION)
```
Comment (knrt10) [inline Makefile:63]: same `--purge` tag

File: Makefile [hunk 3]
```diff
@@ -38,6 +38,30 @@ build-docker-all-push: build-docker-all
 	docker push $(IMAGE_REPOSITORY)/checkpoint:$(VERSION)-amd64
 	docker push $(IMAGE_REPOSITORY)/checkpoint:$(VERSION)-arm64
 
+	docker manifest create $(IMAGE_REPOSITORY)/bootkube:$(VERSION) \
```
Comment (knrt10) [inline Makefile:41]: Can we have all manifest in new make rule? It would be easy to build, annotate and push manifest as a separate command

**Review by knrt10 — CHANGES_REQUESTED** [review]
Some suggestions

File: Makefile (hunk 3 above)
Comment (invidian) [reply Makefile:41]: Make sense.

**Review by knrt10 — APPROVED** [review]

**Pull request merged**